### PR TITLE
Make 'exit;' behave the same as the exit command

### PR DESCRIPTION
**Pull request by damiankloip**
Add an ExitPass so exit; behaves the same as the exit command.

This should be helping us soon in the drush psysh integration (we have problems with the database connection being closed before the drush command has finished, when the shell exits).

### CODE REVIEW

**Comment by bobthecow** [comment]
Nice!

What happens if you do this:

```php
if (false) {
  exit;
} else {
  echo "do something cool!";
}
```

:)

**Comment by GrahamCampbell** [comment]
What's going on with StyleCI here? Does this need rebasing?

**Comment by bobthecow** [comment]
Master is failing on StyleCI, so since this was originally created against master it might be failing as well?

@damiankloip Do you mind rebasing against the dev branch?

**Comment by damiankloip** [comment]
Sorry, forgot to merge develop from upstream.

**Comment by damiankloip** [comment]
@bobthecow Youch! Fair point. Any ideas about the best way to not bail out at the first sniff of exit? :)

**Comment by bobthecow** [comment]
You could probably rewrite the AST to throw an exception on execution, rather than throwing an exception on parse?

**Comment by damiankloip** [comment]
I did try something on leaveNode() but this seems to get out of hand fast..

```php
return new Node\Stmt\Throw_(new Node\Expr\New_(new Node\Name('Psy\Exception\BreakException'), [new Node\Arg(new Node\Scalar\String_('Goodbye.'))]));
```

**Comment by damiankloip** [comment]
@bobthecow Yeah, seems the `ImplicitReturnPass` was an issue. Seems to work much better now. Please give it a try.

**Comment by damiankloip** [comment]
Just updated the tests for the change to using `leaveNode` too.

File: test/Psy/Test/CodeCleaner/ImplicitReturnPassTest.php [hunk 1]
```diff
@@ -33,6 +33,7 @@ public function implicitReturns()
         return array(
             array('4',     'return 4;'),
             array('foo()', 'return foo();'),
+            array('exit()', 'die;'),
```
Comment (damiankloip) [inline test/Psy/Test/CodeCleaner/ImplicitReturnPassTest.php]: Thought it makes sense to also add a new item to the data provider to check that no implicit return is added now.

**Comment by bobthecow** [comment]
Looks good to me. Thanks!

**Pull request merged**